File: C++/Level1/Make PrimeNumber.cpp
/*
	https://programmers.co.kr/learn/courses/30/lessons/12977
*/

#include <vector>
#include <cmath>

using namespace std;

const bool IsFrimeNumber(const int num) {
	int sqrtNum = sqrt(num);

	for (int i = 2; i <= sqrtNum; ++i)
	{
		if (num % i == 0)
		{
			return false;
		}
	}

	return true;
}

int solution(vector<int> nums) {
	int result = 0;

	for (int first = 0; first < nums.size() - 2; ++first)
	{
		for (int second = first + 1; second < nums.size() - 1; ++second)
		{
			for (int third = second + 1; third < nums.size(); ++third)
			{
				result += IsFrimeNumber(nums[first] + nums[second] + nums[third]);
			}
		}
	}

	return result;
}

File: Java/Level1/MakePrimeNumber.java
/*
 * https://school.programmers.co.kr/learn/courses/30/lessons/12977?language=java
 */
public class MakePrimeNumber {
    int IsFrimeNumber(int num) {
        double sqrtNum = Math.sqrt((double)num);
        
        for (int i = 2; i <= sqrtNum; ++i) 
        {
            if (num % i == 0)
            {
                return 0;
            }
        }
    
        return 1;
    }

    public int solution(int[] nums) {
        int result = 0;

	    for (int i = 0; i < nums.length - 2; ++i) 
        {
		    for (int j = i + 1; j < nums.length - 1; ++j)  
            {
			    for (int k = j + 1; k < nums.length; ++k) 
                {
			    	result += IsFrimeNumber(nums[i] + nums[j] + nums[k]);
                }
            }
        }
    
	    return result;
    }
}
